src/services/place_search.rs

```diff
@@ -11,19 +13,123 @@ impl PlaceSearchService {
     pub fn new(client: GooglePlacesClient) -> Self {
         PlaceSearchService { client }
     }
-
     pub async fn nearby_search(
         &self,
-        query: &str,
         location: (f64, f64),
         radius: u32,
+        keyword: Option<&str>,
+        language: Option<Language>,
+        max_price: Option<u32>,
+        min_price: Option<u32>,
+        open_now: Option<bool>,
+        page_token: Option<String>,
+        place_type: Option<HashSet<PlaceTypes>>,
+    ) -> Result<NearbySearchResult, GooglePlacesError> {
+        return
+            self.full_nearby_search(
+                location,
+                Some(radius),
+                keyword,
+                language, 
+                max_price,
+                min_price,
+                open_now,
+                page_token,
+                None,
+                place_type,
+            ).await
+        
+    }
+    pub async fn nearby_search_rank_by_distance(
+        &self,
+        location: (f64, f64),
+        keyword: Option<&str>,
+        language: Option<Language>,
+        max_price: Option<u32>,
+        min_price: Option<u32>,
+        open_now: Option<bool>,
+        page_token: Option<String>,
+        place_type: Option<HashSet<PlaceTypes>>,
+    ) -> Result<NearbySearchResult, GooglePlacesError> {
+        return
+            self.full_nearby_search(
+                location,
+                None,
+                keyword,
+                language,
+                max_price,
+                min_price,
+                open_now,
+                page_token,
+                Some(RankBy::Distance),
+                place_type,
+            ).await
+    }
+
+    async fn full_nearby_search(
+        &self,
+        location: (f64, f64),
+        radius: Option<u32>,
+        keyword: Option<&str>,
+        language: Option<Language>,
+        max_price: Option<u32>,
+        min_price: Option<u32>,
+        open_now: Option<bool>,
+        page_token: Option<String>,
+        rank_by: Option<RankBy>,
+        place_type: Option<HashSet<PlaceTypes>>,
     ) -> Result<NearbySearchResult, GooglePlacesError> {
         // Construct the request URL
         // "{}/maps/api/place/nearbysearch/json?keyword={}&location={}%2c{}&radius={}&key={}",
-        let url = format!(
-            "https://maps.googleapis.com/maps/api/place/nearbysearch/json?location={},{}&radius={}&keyword={}&key={}",
-            location.0, location.1, radius, query, self.client.get_api_key()
+        let mut url = format!(
+            "https://maps.googleapis.com/maps/api/place/nearbysearch/json?location={}%2C{}&key={}",
+            location.0,location.1, self.client.get_api_key()
         );
+        // Radius
```

Comment:
get rid of these comments, maybe just add one saying you're pushing optional params onto this str?